src/programmatic_policy_learning/approaches/lpp_approach.py

```diff
@@ -85,17 +132,78 @@ def get_program_set(
         env_spec=env_specs if env_specs is not None else {},
         start_symbol=start_symbol,
     )
+    return program_generator, dsl_dict
 
-    logging.info(f"Generating {num_programs} programs")
 
+def _generate_with_dsl_generator(
+    program_generation: dict[str, Any],
+    env_specs: dict[str, Any] | None,
+    start_symbol: int,
+) -> tuple[GrammarBasedProgramGenerator, dict[str, Any]]:
+    """Generate programs using the DSL generator."""
+
+    cache_path = Path(tempfile.NamedTemporaryFile(suffix=".db").name)
+    cache = SQLite3PretrainedLargeModelCache(cache_path)
+    llm_client = OpenAIModel("gpt-4o-mini", cache)
+    prompt_path = program_generation.get("dsl_generator_prompt", "")
+    with open(
+        prompt_path,
+        "r",
+        encoding="utf-8",
+    ) as file:
+        prompt = file.read()
+
+    object_types = ["tpn.EMPTY", "tpn.TOKEN", "None"]
```

Comment:
everything else in this file looks pretty general so I am a bit sus of this line